Softeq.XToolkit.ProjectInfrastructure/HttpResponce.cs

```diff
@@ -16,6 +17,22 @@ public class HttpResponse
         public IEnumerable<KeyValuePair<string, IEnumerable<string>>> ContentHeaders { get; set; }
         public DateTimeOffset? Expires { get; set; }
 
+        public object Clone()
+        {
+            return new HttpResponse
+            {
+                Content = this.Content,
+                BinaryContent = this.BinaryContent?.Clone() as byte[],
+                IsSuccessful = this.IsSuccessful,
+                StatusCode = this.StatusCode,
+                IsNoInternet = this.IsNoInternet,
+                ResponseUri = this.ResponseUri,
+                Headers = this.Headers?.ToList(),
+                ContentHeaders = this.ContentHeaders?.ToList(),
+                Expires = this.Expires
+            };
```

Comment:
```suggestion
            return this.MemberwiseClone();
```

Reply:
MemberwiseClone is copying collections references, but is doesn't create new collection (ContentHeaders and BinaryContent).